wandb/sdk/launch/agent/agent.py

```diff
@@ -227,22 +254,15 @@ def loop(self) -> None:
                                     f"Error running job: {traceback.format_exc()}"
                                 )
                                 self._api.ack_run_queue_item(job["runQueueItemId"])
-                for job_id in self.job_ids:
-                    self._update_finished(job_id)
                 if self._ticks % 2 == 0:
-                    if self._running == 0:
+                    if len(self._jobs.keys()) == 0:
                         self.update_status(AGENT_POLLING)
                     else:
                         self.update_status(AGENT_RUNNING)
                     self.print_status()
                 time.sleep(AGENT_POLLING_INTERVAL)
 
         except KeyboardInterrupt:
-            # temp: for local, kill all jobs. we don't yet have good handling for different
-            # types of runners in general
-            for _, run in self._jobs.items():
-                if isinstance(run, LocalSubmittedRun):
-                    run.command_proc.kill()
```

Comment:
What happens now when you stop the agent? How do we kill those runs?

Reply:
I wasn't sure how to get around this at the time of commit, but I think adding an `except KeyboardInterrupt` clause in the thread function should let us kill it from there. Will add in next revision.